chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/RemovedCriteriaComparisonReviewer.java

```diff
@@ -0,0 +1,423 @@
+            updatedListing.getErrorMessages().add(
+                    msgUtil.getMessage("listing.removedCriteriaEditNotAllowed", updatedCert.getNumber(),
+                            "Privacy and Security Framework"));
+        }
+        if (isSedChanged(existingCert, updatedCert)) {
+            updatedListing.getErrorMessages().add(
+                    msgUtil.getMessage("listing.removedCriteriaEditNotAllowed", updatedCert.getNumber(), "SED"));
+        }
+    }
+
+    private boolean isGapChanged(CertificationResult existingCert, CertificationResult updatedCert) {
+        if (certResultRules.hasCertOption(updatedCert.getNumber(), CertificationResultRules.GAP)) {
+            return !ObjectUtils.equals(existingCert.isGap(), updatedCert.isGap());
+        }
+        return false;
+    }
+
+    private boolean isG1SuccessChanged(CertificationResult existingCert, CertificationResult updatedCert) {
+        if (certResultRules.hasCertOption(updatedCert.getNumber(), CertificationResultRules.G1_SUCCESS)) {
+            return !ObjectUtils.equals(existingCert.isG1Success(), updatedCert.isG1Success());
+        }
+        return false;
+    }
+
+    private boolean isG2SuccessChanged(CertificationResult existingCert, CertificationResult updatedCert) {
+        if (certResultRules.hasCertOption(updatedCert.getNumber(), CertificationResultRules.G2_SUCCESS)) {
+            return !ObjectUtils.equals(existingCert.isG2Success(), updatedCert.isG2Success());
+        }
+        return false;
+    }
+
+    private boolean isAdditionalSoftwareChanged(CertificationResult existingCert, CertificationResult updatedCert) {
+        if (certResultRules.hasCertOption(updatedCert.getNumber(), CertificationResultRules.ADDITIONAL_SOFTWARE)) {
+            for (CertificationResultAdditionalSoftware updatedAdditionalSoftware : updatedCert.getAdditionalSoftware()) {
+                boolean hasMatch = false;
+                for (CertificationResultAdditionalSoftware existingAdditionalSoftware : existingCert.getAdditionalSoftware()) {
+                    if (updatedAdditionalSoftware.matches(existingAdditionalSoftware)) {
+                        hasMatch = true;
+                    }
+                }
+                if (!hasMatch) {
+                    return true;
+                }
+            }
+
+            for (CertificationResultAdditionalSoftware existingAdditionalSoftware : existingCert.getAdditionalSoftware()) {
+                boolean hasMatch = false;
+                for (CertificationResultAdditionalSoftware updatedAdditionalSoftware : updatedCert.getAdditionalSoftware()) {
+                    if (existingAdditionalSoftware.matches(updatedAdditionalSoftware)) {
+                        hasMatch = true;
+                    }
+                }
+                if (!hasMatch) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
```

Comment:
I think in the methods like this, if you tried to use some of the Java 8 stream methods, it would greatly improve the readability of the code.  I think the Stream.findAny with the Optional<T> type would work well here.